standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -3705,12 +3707,22 @@ public CompactionResponse compact(CompactionRequest rqst) throws MetaException {
 
         long id = generateCompactionQueueId(stmt);
 
+        final ValidCompactorWriteIdList tblValidWriteIds = TxnUtils.createValidCompactWriteIdList(
+            getValidWriteIds(new GetValidWriteIdsRequest(
```

Comment:
could we refactor this a bit, at least move GetValidWriteIdsRequest into a separate declaration